### PR TITLE
Java wrapper for xgboost

### PR DESCRIPTION
**Pull request by yanqingmen**
hi @tqchen 

the third PR ~ 
I hope this time I've made things better.

modifications:
1. examples are added in xgboost4j-demo, the structure are same with demo/guide-python

### CODE REVIEW

File: java/xgboost4j/src/main/java/org/dmlc/xgboost4j/util/NativeUtils.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
```
Comment (tqchen) [inline java/xgboost4j/src/main/java/org/dmlc/xgboost4j/util/NativeUtils.java]: change this license header?

File: java/xgboost4j/src/main/resources/lib/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+please put native library in this package.
```
Comment (tqchen) [inline java/xgboost4j/src/main/resources/lib/README.md]: can we auto generate the lib to this path? or let native lib search for possible path? see https://github.com/dmlc/xgboost/blob/master/wrapper/xgboost.py#L58

File: java/xgboost4j/src/main/java/org/dmlc/xgboost4j/util/TransferUtil.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package org.dmlc.xgboost4j.util;
+
+import org.dmlc.xgboost4j.DMatrix;
+
+/**
+ *
```
Comment (tqchen) [inline java/xgboost4j/src/main/java/org/dmlc/xgboost4j/util/TransferUtil.java]: this util seems to be private and should be invisible to the user. Consider put the two functions in private DMatrix or Trainer?

File: java/xgboost4j/src/main/java/org/dmlc/xgboost4j/util/Trainer.java [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+     * @param params Booster params.
+     * @param dtrain Data to be trained.
+     * @param round Number of boosting iterations.
+     * @param evalMats Data to be evaluated (may include dtrain)
+     * @param evalNames name of data (used for evaluation info)
+     * @param obj customized objective (set to null if not used)
+     * @param eval customized evaluation (set to null if not used)
+     * @return trained booster
+     */
+    public static Booster train(Params params, DMatrix dtrain, int round,
```
Comment (tqchen) [inline java/xgboost4j/src/main/java/org/dmlc/xgboost4j/util/Trainer.java]: user Iterable<Entry<String, String> > for params?
Ideally it should be Iterable <Entry<String, anytype that have toString> >

File: java/xgboost4j/src/main/java/org/dmlc/xgboost4j/util/Trainer.java [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+            DMatrix[] evalMats, String[] evalNames, IObjective obj, IEvaluation eval) {
```
Comment (tqchen) [inline java/xgboost4j/src/main/java/org/dmlc/xgboost4j/util/Trainer.java]: similarly, evalMats and evalNames might be combined, for example
```Iterable<Entry<String, DMatrix> >``` might be more convenient for users 

see also watchlist in https://github.com/dmlc/xgboost/blob/master/wrapper/xgboost.py#L639

**Comment by tqchen** [comment]
Thanks!, This was much better. I made a few comments on the code. Please make this fix and I will merge it in. 

Thanks for the great effort!

**Comment by yanqingmen** [comment]
Ok, I'll make this fix.

**Comment by yanqingmen** [comment]
hi, @tqchen 

I've made some modification:
1. rm TransferUtil and make the two funcs private
2. change Params to Iterable<Entry<String, Object>>
3. add a "create_wrap.bat" to help windows user to set the native library properly
4. add a WachList class to handle evaluation data pairs

**Comment by tqchen** [comment]
I think we do not need WatchList class, simply ```Iterable<Entry<String, DMatrix>>``` will do. This makes things as generic as possible

**Comment by yanqingmen** [comment]
ok, I'll change it to Iterable\<Entry\<String, DMatrix\>\>

**Comment by yanqingmen** [comment]
I've done this

**Comment by tqchen** [comment]
Thanks! this is now in the master branch

**Pull request merged**